pkg/extension/sumologicextension/credentials.go

```diff
@@ -0,0 +1,181 @@
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/sumologicextension/api"
+	"go.uber.org/zap"
+)
+
+// Authentication is an interface to get collector authentication data
+type CredsGetter interface {
+	CheckCollectorCredentials() bool
+	GetStoredCredentials() (api.OpenRegisterResponsePayload, error)
+	RegisterCollector() (api.OpenRegisterResponsePayload, error)
+}
+
+// Authenticator is a common structure for both types of getting collector credentials
+// stored in file and registration new collector.
+type credsGetter struct {
+	conf   *Config
+	logger *zap.Logger
+}
+
+// CheckCollectorCredentials checks if collector credentials can be found in path
+// configured in the config.
+func (cr credsGetter) CheckCollectorCredentials() bool {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return false
+	}
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
+// GetStoredCredentials retrieves collector credentials stored in local file system and then decrypts it
+// using hashed collector name as passphrase.
+func (cr credsGetter) GetStoredCredentials() (api.OpenRegisterResponsePayload, error) {
+	filenameHash, err := hash(cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	path := path.Join(cr.conf.CollectorCredentialsPath, filenameHash)
+	creds, err := os.Open(path)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	defer creds.Close()
+	encryptedCreds, err := ioutil.ReadAll(creds)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	collectorCreds, err := decrypt(encryptedCreds, cr.conf.CollectorName)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	var credentialsInfo api.OpenRegisterResponsePayload
+	if err = json.Unmarshal(collectorCreds, &credentialsInfo); err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+
+	return credentialsInfo, nil
+}
+
+// RegisterCollector registers new collector using registration API and returns collector credentials.
+func (cr credsGetter) RegisterCollector() (api.OpenRegisterResponsePayload, error) {
+	baseUrl := strings.TrimSuffix(cr.conf.ApiBaseUrl, "/")
+	u, err := url.Parse(baseUrl)
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, err
+	}
+	u.Path = registerUrl
+
+	// TODO: just plain hostname or we want to add some custom logic when setting
+	// hostname in request?
+	hostname, err := os.Hostname()
+	if err != nil {
+		return api.OpenRegisterResponsePayload{}, fmt.Errorf("cannot get hostname: %w", err)
+	}
+
+	var buff bytes.Buffer
+	if err = json.NewEncoder(&buff).Encode(api.OpenRegisterRequestPayload{
+		Ephemeral:     true, // TODO: change that
+		CollectorName: cr.conf.CollectorName,
+		Description:   cr.conf.CollectorDescription,
+		Category:      cr.conf.CollectorCategory,
+		Hostname:      hostname,
```

Comment:
update this to what was changed on `main` in https://github.com/SumoLogic/sumologic-otel-collector/pull/39 and https://github.com/SumoLogic/sumologic-otel-collector/pull/38